pkg/component/observability/monitoring/kubestatemetrics/resources.go

```diff
@@ -736,3 +712,14 @@ func (k *kubeStateMetrics) nameSuffix() string {
 	}
 	return suffix + k.values.NameSuffix
 }
+
+func (k *kubeStateMetrics) customResourceStateConfigMap() *corev1.ConfigMap {
+	customResourceStateConfig, err := yaml.Marshal(NewCustomResourceStateConfig())
+	utilruntime.Must(err)
```

Comment:
Yes, you're right. I didn't return the error in the first place because the configuration that could error here is statically generated. But it definitely feels better to return the error and be on the safe-side 👍🏻